### PR TITLE
Saucelabs in Docker 

### PR DESCRIPTION
**Pull request by glogiotatidis**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM python:2
+
+WORKDIR /app
+
+ENV BASE_URL "https://www.mozilla.org"
```
Comment (davehunt) [inline Dockerfile]: The default base URL should probably not be production, can we set this to dev?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ENV BROWSER_NAME Firefox
+ENV BROWSER_VERSION 39.0
+ENV PLATFORM "Windows 7"
+ENV SAUCE_CREDENTIALS_PATH /tmp/creds
+ENV BUILD_TAG docker
```
Comment (davehunt) [inline Dockerfile]: The build tag should be unique for each build, are we setting this somewhere else?

File: run.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+py.test -r=fsxXR --verbose -n 15 \
+        --baseurl=${BASE_URL} \
+        --browsername="${BROWSER_NAME}" \
+        --browserver=${BROWSER_VERSION} \
+        --platform="${PLATFORM}" \
+        --junitxml=results/results.xml \
+        --saucelabs=${SAUCE_CREDENTIALS_PATH} \
+        --capability="selenium-version:${SELENIUM_VERSION}" \
```
Comment (davehunt) [inline run.sh]: Is a default Selenium version being set anywhere?

**Comment by davehunt** [comment]
Thanks @glogiotatidis, sorry if my comments are a little naive - I'm not too familiar with Docker yet. So as I understand it, we have a mozorg docker repository, and we're creating a mcom-tests container image? Do I need to do anything to configure my machine for this docker repo?

File: README.md [hunk 4]
```diff
@@ -49,6 +57,48 @@ To run tests locally, it's a simple case of calling the command below from this
...
+[Dockerfile](Dockerfile) and can be overriden with environment
+variables. For example to test against the stage server, override
+BASE_URL environment variable:
+
+    docker run -v `pwd`/creds.yml:/tmp/creds -e BASE_URL=https://www.allizom.org mozorg/mcom-tests
+
+__note__
+
+Docker registry builds new docker images for every code commit in
+mozilla/mcom-tests. To an up to date image run:
```
Comment (davehunt) [inline README.md]: To <get> an up to date image?

File: Dockerfile (hunk 1 above)
Comment (glogiotatidis) [inline Dockerfile]: kk

File: Dockerfile (hunk 2 above)
Comment (glogiotatidis) [inline Dockerfile]: In Jenkins were we're using this, yes we do set a build_tag based on the jenkins job name https://github.com/mozilla/bedrock/pull/3401/files#diff-ca92073a5860ceb5098b7f64e3cd5cadR11 which is unique. 

You can use the `-e BUILD_TAG=foo` docker run flag to set it in other runs.

Does that sound good enough?

Comment (davehunt) [inline Dockerfile]: Yep, sounds great. Thanks!

File: run.sh (hunk 3 above)
Comment (glogiotatidis) [inline run.sh]: Nice catch, now set as an ENV var.

**Comment by glogiotatidis** [comment]
> Thanks @glogiotatidis, sorry if my comments are a little naive - I'm not too familiar with Docker yet. So as I understand it, we have a mozorg docker repository, and we're creating a mcom-tests container image? Do I need to do anything to configure my machine for this docker repo?

All great points, nothing to worry about. 

We need to setup an Automated Build in the [docker registry](https://registry.hub.docker.com/) as soon as we merge this in. I've run all my tests with the `giorgos/mcom-tests` image which builds of my repository. So to test this in your machine meanwhile just replace `mozorg/mcom-tests` with `giorgos/mcom-tests`.  If you have docker running there shouldn't be anything special about this image.

**Pull request merged**